### ISSUE TITLE
[Bug] Calculation of AWS Lambda estimate using cloud usage API is incorrect

### ISSUE THREAD

**Issue by millasvs** [cloud-carbon-footprint/cloud-carbon-footprint#624]
From the file aws/src/lib/Lambda.ts:

```   
return filteredResults.map((resultByDate) => {
      const timestampField = resultByDate[0]
      const wattsField = resultByDate[1]
      const timestamp = new Date(timestampField.value.substr(0, 10))
      const wattHours =
        Number.parseFloat(wattsField.value) * AWS_CLOUD_CONSTANTS.getPUE()
      const co2e = estimateCo2(
        wattHours,
        region,
        AWS_EMISSIONS_FACTORS_METRIC_TON_PER_KWH,
      )
      return {
        timestamp,
        kilowattHours: wattHours,
        co2e,
      }
    })
```

The number of watt hours are being reported as kilowatt hours.

I.e. the results for the AWS Lambda emissions are off by a factor of 1000. 

I started looking at this because the estimates for the emissions of my team's AWS Lambda instances seemed impossibly high, and much, much higher than other services.

Of course, you could then say that the variable above is simply called wattHours even though it is actually kilowatt hours. But looking at the way this variable is calculated it seems to me to be watt hours, not kilowatt hours.

This is the way the watt hours are being extracted from Cloudwatch:

```
  private runQuery = async (
    start: Date,
    end: Date,
    groupNames: string[],
  ): Promise<string> => {
    const averageWatts =
      AWS_CLOUD_CONSTANTS.MIN_WATTS_AVG +
      (AWS_CLOUD_CONSTANTS.AVG_CPU_UTILIZATION_2020 / 100) *
        (AWS_CLOUD_CONSTANTS.MAX_WATTS_AVG - AWS_CLOUD_CONSTANTS.MIN_WATTS_AVG)

    const query = `
            filter @type = "REPORT"
            | fields datefloor(@timestamp, 1d) as Date, @duration/1000 as DurationInS, @memorySize/1000000 as MemorySetInMB, ${averageWatts} * DurationInS/3600 * MemorySetInMB/1792 as wattsPerFunction
            | stats sum(wattsPerFunction) as Watts by Date 
            | sort Date asc`

    const params = {
      startTime: start.getTime(),
      endTime: end.getTime(),
      queryString: query,
      logGroupNames: groupNames,
    }

    const queryData = await this.serviceWrapper.startCloudWatchLogsQuery(params)

    return queryData.queryId
  }
```

Am I going crazy or is what is calculated here just watt hours, not kilowatt hours?

We are taking the average watts (W), multiplying by duration in hours (h) and the number of vCPUs (as estimated by looking at the memory usage). Nowhere is it divided by 1000 to actually be kWh.

So in other words, the results when estimating AWS Lambda emissions using the cloud usage API method is incorrect by a factor of a 1000.

**Comment by ericksod** [comment]
This has been resolved here: https://github.com/cloud-carbon-footprint/cloud-carbon-footprint/commit/4d8653221ea5472ef59d7772cb7b4648a9ca8640.
Closing this issue!
